src/verifier/outcome.rs

```diff
@@ -0,0 +1,156 @@
+use std::{collections::HashMap, fmt};
+
+pub type Result<T, E = Failure> = ::std::result::Result<T, E>;
+
+/// The outcome of attempting to verify a credential.
+#[derive(uniffi::Enum)]
+pub enum Outcome {
+    /// The credential was successfully verified.
+    Verified { credential_info: CredentialInfo },
+    /// The credential could not be verified.
+    Unverified {
+        credential_info: Option<CredentialInfo>,
+        failure: Failure,
+    },
+}
+
+/// Information about the verified credential.
+#[derive(uniffi::Record)]
+pub struct CredentialInfo {
+    /// The credential title that should be displayed on the success screen.
+    pub title: String,
+    /// The image that should be displayed on the success screen.
+    pub image: Vec<u8>,
+    /// The claims decoded from the credential.
+    pub claims: HashMap<String, ClaimValue>,
+}
+
+/// Credential claim values.
+#[derive(uniffi::Enum, Debug)]
+pub enum ClaimValue {
+    /// Any text claim that doesn't need special formatting.
+    Text { value: String },
+    /// A date claim in the format `[year]-[month]-[day]`.
+    Date { value: String },
+    /// MapArray
+    MapClaim { value: HashMap<String, String> },
+}
+
+#[derive(uniffi::Record, Debug)]
+/// A verification failure with a code and reason.
+pub struct Failure {
```

Comment:
I've added a note to change this to an `#[uniffi::Error]` type. Given the current implementation, we're going to wait to merge in.

I think it will make sense to make this change once we get the monorepo established and can see the integration.